source/Calamari.AzureResourceGroup/LegacyDeployAzureResourceGroupBehaviour.cs

```diff
@@ -72,21 +73,25 @@ public async Task Execute(RunningDeployment deployment)
 
             // We re-create the client each time it is required in order to get a new authorization-token. Else, the token can expire during long-running deployments.
             Func<Task<IResourceManagementClient>> createArmClient = async () =>
-                                                              {
-                                                                  var token = !jwt.IsNullOrEmpty()
-                                                                      ? await new AzureOidcAccount(variables).Credentials(CancellationToken.None)
-                                                                      : await new AzureServicePrincipalAccount(variables).Credentials();
-                                                                  var resourcesClient = new ResourceManagementClient(token, AuthHttpClientFactory.ProxyClientHandler())
-                                                                  {
-                                                                      SubscriptionId = subscriptionId,
-                                                                      BaseUri = new Uri(resourceManagementEndpoint),
-                                                                  };
-                                                                  resourcesClient.HttpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + token);
-                                                                  resourcesClient.HttpClient.BaseAddress = new Uri(resourceManagementEndpoint);
-                                                                  return resourcesClient;
-                                                              };
-
-            await CreateDeployment(createArmClient, resourceGroupName, deploymentName, deploymentMode, template, parameters);
+                                                                    {
+                                                                        var account = AzureAccountFactory.Create(variables);
+                                                                        var token = new TokenCredentials(await account.GetAuthorizationToken(log, CancellationToken.None));
```

Comment:
Consolidating around the `AzureAccountFactory` and the new `IAzureAccount.GetAuthorizationToken` method